editor/src/core/shared/set-utils.ts

```diff
@@ -29,3 +29,19 @@ export function intersection<T>(sets: Array<Set<T>>): Set<T> {
 
   return rest.reduce((acc, v) => new Set([...acc].filter((x) => v.has(x))), first)
 }
+
+// Values that are in the first set but not in the second set.
+export function difference<T>(firstSet: Set<T>, secondSet: Set<T>): Set<T> {
+  let result = new Set(firstSet)
+  secondSet.forEach((value) => result.delete(value))
+  return result
+}
+
+export function getSingleValueOnly<T>(set: Set<T>): T {
+  if (set.size === 1) {
+    for (const value of set) {
+      return value
+    }
+  }
+  throw new Error(`Set had ${set.size} when it was expected to have just 1.`)
```

Comment:
Why is this an error and not `null`? As far as I can see, none of the call sites actually checks whether `set` is a singleton, and this way the inner workings of this function leak out in a really unpredictable way

Reply:
I disagree, this is being used to ensure that the sets at the call sites _are_ singletons and then retrieve the single value within. If It were called `getFirst` I would be more inclined to agree.